metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/CatalogTableDataset.java

```diff
@@ -14,12 +14,12 @@
 public class CatalogTableDataset implements SparkDataset {
   private final DatasetUrn urn;
 
-  public CatalogTableDataset(CatalogTable table) {
-    this(table.qualifiedName());
+  public CatalogTableDataset(CatalogTable table, String platformInstance, FabricType fabricType) {
+    this(table.qualifiedName(), platformInstance, fabricType);
   }
 
-  public CatalogTableDataset(String dsName) {
-    this.urn = new DatasetUrn(new DataPlatformUrn("hive"), dsName, FabricType.PROD);
+  public CatalogTableDataset(String dsName, String platformInstance, FabricType fabricType) {
+    this.urn = LineageUtils.createDatasetUrn("hive", platformInstance, dsName, fabricType);
```

Comment:
This could be pushed up to the interface level with a call to super(<nameString>)

Reply:
Done